cmd/kube-apiserver/app/server.go

```diff
@@ -547,9 +549,55 @@ func BuildAuthenticator(s *options.ServerRunOptions, storageFactory serverstorag
 	return authenticatorConfig.New()
 }
 
+// buildWebhookDialer constructs a custom dialer function for the authorization webhook
+// that knows how to resolve references to Kubernetes services (of the form "servicename.namespace.svc")
+// to the IP address of the endpoint that currently implements it.
+func buildWebhookDialer(
+	serviceResolver aggregatorapiserver.ServiceResolver,
+	proxyTransport *http.Transport,
+) func(network, address string) (net.Conn, error) {
+	// Following sample code from issue #54163 by liggitt@redhat.com.
+	baseDialer := net.Dial
+	if proxyTransport != nil && proxyTransport.Dial != nil {
+		baseDialer = proxyTransport.Dial
+	}
+
+	return func(network, address string) (net.Conn, error) {
+		host, _, err := net.SplitHostPort(address)
+		if err != nil {
+			glog.V(6).Infof("while splitting hostport: %v", err)
+			return baseDialer(network, address)
+		}
+		// "servicename.namespace.svc"
+		if !strings.HasSuffix(host, ".svc") {
+			return baseDialer(network, address)
+		}
+		segments := strings.Split(host, ".")
+		if len(segments) != 3 {
+			glog.V(6).Infof("Dialer address is not a Kubernetes svc reference: %v", host)
```

Comment:
Do we need this log output? Doesn't look like an error.

Reply:
ends in .svc but is not a valid kubernetes reference... it's unexpected, I guess